src/corehost/common/pal.unix.cpp

```diff
@@ -20,6 +21,17 @@
 #define symlinkEntrypointExecutable "/proc/curproc/exe"
 #endif
 
+bool pal::ensure_file(const pal::string_t& src, const pal::string_t& dest)
```

Comment:
If this is present only for mscorlib patching, lets have a TODO to remove this as well. And given that it is temporary, I would suggest we keep the behaviour consistent and copy on non-Windows as well (eventually, it will be a non-symlinked mscorlib.ni.dll).